File: src/core/slices/station.ts
import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { RootState } from "..";

export interface IStation {
  latitude: number;
  longitude: number;
  speed: number;
  altitude: number;
  planetRadius: number;
  angle: number;
  planetName: string;
  solarPanelStatus: string;
  scientificInstrumentsStatus: string;
  navigationSystemStatus: string;
  temperature: number;
  hullStatus: string;
  fuelLevel: number;
}

const initialState: IStation = {
  latitude: 0,
  longitude: 0,
  speed: 0,
  altitude: 0,
  planetRadius: 0,
  angle: 0,
  planetName: "",
  scientificInstrumentsStatus: "",
  navigationSystemStatus: "",
  solarPanelStatus: "",
  temperature: 0,
  hullStatus: "",
  fuelLevel: 0,
};

const slice = createSlice({
  name: "stationSlice",
  initialState,
  reducers: {
    updateInfo: (state, action: PayloadAction<IStation>) => {
      state.latitude = action.payload.latitude || 0;
      state.longitude = action.payload.longitude || 0;
      state.speed = action.payload.speed || 0;
      state.altitude = action.payload.altitude || 0;
      state.planetName = action.payload.planetName || "";
      state.angle = action.payload.angle || 0;
      state.solarPanelStatus = action.payload.solarPanelStatus || "";
      state.planetName = action.payload.planetName || "";
      state.scientificInstrumentsStatus = action.payload.scientificInstrumentsStatus || "";
      state.navigationSystemStatus = action.payload.navigationSystemStatus || "";
      state.hullStatus = action.payload.hullStatus || "";
      state.temperature = action.payload.temperature || 0;
      state.fuelLevel = action.payload.fuelLevel || 0;
    },
  },
});

export const { updateInfo } = slice.actions;

export const selectStationInfo = (state: RootState) => state.station;

export const { reducer } = slice;

File: backend/serviceStation/internal/app/ds/station.go
package ds

import (
	"encoding/json"
	"io/ioutil"
	"log"
	"os"
	"sync"
)

// Location представляет данные о местоположении станции.
type Location struct {
	Latitude                  float64 `json:"latitude"`
	Longitude                 float64 `json:"longitude"`
	Speed                     float64 `json:"speed"`
	Altitude                  float64 `json:"altitude"`
	PlanetRadius              int     `json:"planetRadius"`
	Angle                     float64 `json:"angle"`
	PlanetName                string  `json:"planetName"`
	SolarPanelStatus          string  `json:"solarPanelStatus"`
	FuelLevel                 float64 `json:"fuelLevel"`
	HullStatus                string  `json:"hullStatus"`
	Temperature               float64 `json:"temperature"`
	ScientificInstrumentsStatus string  `json:"scientificInstrumentsStatus"`
	NavigationSystemStatus    string  `json:"navigationSystemStatus"`
}

var mu sync.Mutex
const stationDataFile = "../station.json"

func ReadLocationFromFile() (*Location, error) {
    mu.Lock()
    defer mu.Unlock()

    file, err := os.Open(stationDataFile)
    if err != nil {
		log.Println("sdfsdfsdf")
        return nil, err
    }
    defer file.Close()

    data, err := ioutil.ReadAll(file)
    if err != nil {
        return nil, err
    }

    var location Location
    err = json.Unmarshal(data, &location)
    if err != nil {
        return nil, err
    }

    return &location, nil
}
func WriteLocationToFile(location *Location) error {
	mu.Lock()
	defer mu.Unlock()

	data, err := json.Marshal(location)
	if err != nil {
		return err
	}

	// Попытайтесь создать файл, если он не существует
	file, err := os.OpenFile(stationDataFile, os.O_WRONLY|os.O_CREATE, 0644)
	if err != nil {
		return err
	}
	defer file.Close()

	err = ioutil.WriteFile(stationDataFile, data, 0644)
	if err != nil {
		return err
	}

	return nil
}
